src/core/control/RecentManager.cpp

```diff
@@ -245,12 +259,43 @@ void RecentManager::updateMenu() {
 
     auto base = insert_items_job(xoj_files, 0);
 
-    GtkWidget* separator = gtk_separator_menu_item_new();
-    gtk_menu_shell_append(GTK_MENU_SHELL(menu), separator);
-    gtk_widget_set_visible(GTK_WIDGET(separator), true);
-    this->menuItemList.push_back(separator);
+    if (!listIsEmpty(recentManager)) {
+        GtkWidget* separator = gtk_separator_menu_item_new();
+        gtk_menu_shell_append(GTK_MENU_SHELL(menu), separator);
+        gtk_widget_set_visible(GTK_WIDGET(separator), true);
+        this->menuItemList.push_back(separator);
+    }
 
     insert_items_job(pdf_files, base);
 
+    if (!listIsEmpty(recentManager)) {
```

Comment:
Those successive calls to listIsEmpty implies the list of recent files is iterated upon several times, and unnecessarily so. Note that `listIsEmpty(...) == false` if and only if `pdf_files.empty() && xopp_files.empty()`.
I guess a more optimal code would simply add
```
if (pdf_files.empty() && xopp_files.empty()) {
    g_list_free_full(items, GDestroyNotify(gtk_recent_info_unref));
    return;
}
```
right after the loop populating the vectors pdf_files and xopp_files.

With this, `listIsEmpty()` becomes unused and can be removed as well.